src/GafferScene/RendererAlgo.cpp

```diff
@@ -1121,18 +1121,49 @@ struct CameraOutput : public LocationOutput
 		const size_t cameraMatch = m_cameraSet.match( path );
 		if( cameraMatch & IECore::PathMatcher::ExactMatch )
 		{
-			IECore::ConstObjectPtr object = scene->objectPlug()->getValue();
-			if( const Camera *camera = runTimeCast<const Camera>( object.get() ) )
+			// Sample cameras and apply globals
+
+			vector<ConstObjectPtr> samples; vector<float> sampleTimes;
+			RendererAlgo::objectSamples( scene, deformationSegments(), shutter(), samples, sampleTimes );
+
+			vector<ConstCameraPtr> cameraSamples; cameraSamples.reserve( samples.size() );
+			for( const auto &sample : samples )
 			{
-				IECoreScene::CameraPtr cameraCopy = camera->copy();
+				if( auto cameraSample = runTimeCast<const Camera>( sample.get() ) )
+				{
+					IECoreScene::CameraPtr cameraSampleCopy = cameraSample->copy();
+					GafferScene::RendererAlgo::applyCameraGlobals( cameraSampleCopy.get(), m_globals, scene );
+					cameraSamples.push_back( cameraSampleCopy );
+				}
+			}
 
-				GafferScene::RendererAlgo::applyCameraGlobals( cameraCopy.get(), m_globals, scene );
+			// Create ObjectInterface
 
-				IECoreScenePreview::Renderer::ObjectInterfacePtr objectInterface = renderer()->camera(
-					name( path ),
-					cameraCopy.get(),
-					attributesInterface().get()
-				);
+			if( cameraSamples.size() && cameraSamples.size() == samples.size() )
```

Comment:
Should there be some sort of warning if this fails?  That would correspond to something like a camera that exists on frame, but not at shutter open?  Seems kinda confusing to fail silently

Reply:
Added in d9d1135.